src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,128 @@
+        String current = this.ui.nextLine();
+
+        while (!current.equals("bye")) {
+            try {
+                String information = Parser.parseCommand(current);
+                String[] parts = information.split("\\|");
+
+                switch (parts[0]) {
+                case "list":
+                    this.ui.printTasks(this.tasks);
+                    break;
+                case "mark":
+                    Task marked = this.tasks.changeTaskStatus(Integer.parseInt(parts[1]), true);
+                    this.ui.printChangeTaskStatus(marked, true);
+                    break;
+                case "unmark":
+                    Task unmarked = this.tasks.changeTaskStatus(Integer.parseInt(parts[1]), false);
+                    this.ui.printChangeTaskStatus(unmarked, false);
+                    break;
+                case "delete":
+                    Task deleted = this.tasks.removeTask(Integer.parseInt(parts[1]));
+                    this.ui.printDeleteTask(deleted, this.tasks);
+                    break;
+                case "todo":
+                    Task todo = new Todo(parts[1]);
+                    tasks.addTask(todo);
+                    this.ui.printAddTask(todo, this.tasks);
+                    break;
+                case "deadline":
+                    Task deadline = new Deadline(
+                            parts[1],
+                            LocalDateTime.parse(parts[2], dateTimeFormatter)
+                    );
+                    tasks.addTask(deadline);
+                    this.ui.printAddTask(deadline, this.tasks);
+                    break;
+                case "event":
+                    Task event = new Event(
+                            parts[1],
+                            LocalDateTime.parse(parts[2], dateTimeFormatter),
+                            LocalDateTime.parse(parts[3], dateTimeFormatter)
+                    );
+                    tasks.addTask(event);
+                    this.ui.printAddTask(event, this.tasks);
+                    break;
+                case "find":
+                    ArrayList<Task> targetTasks = this.tasks.findMatchTasks(parts[1]);
+                    this.ui.printMatchTask(targetTasks);
+                    break;
+                default:
+                    throw new DukeException("I'm sorry but I don't know what that means!");
+                }
+            } catch (DukeException e) {
+                this.ui.printDukeException(e);
+            } catch (DateTimeParseException e) {
+                this.ui.printDateTimeParseException();
+            }
+
+            current = this.ui.nextLine();
+        }
+
+        try {
+            this.storage.writeTasksToStorage(this.tasks);
+        } catch (IOException e) {
+            this.ui.printIoException(e);
+        }
+
+        this.ui.printDukeClosing();
+        this.ui.closeScanner();
+        System.exit(0);
```

Comment:
I do not really prefer the switch statements approach as it makes the code really long. One suggestion is to make a class named command class to abstract out the cases perhaps?